src/Service.GraphQLBuilder/Mutations/CreateMutationBuilder.cs

```diff
@@ -42,31 +48,168 @@ public static class CreateMutationBuilder
+            // The complex fields (for related entities) would be added later when we return from recursion.
+            // Adding the input object to the dictionary ensures that we don't go into infinite recursion and return whenever
+            // we find that the input object has already been created for the entity.
+            inputs.Add(input.Name, input);
+
+            // Generate fields for related entities only if nested mutations are supported for the database flavor.
+            if (DoesRelationalDBSupportNestedCreate(databaseType))
+            {
+                // 2. Complex input fields.
+                // Evaluate input objects for related entities.
+                IEnumerable<InputValueDefinitionNode> complexInputFields =
+                    objectTypeDefinitionNode.Fields
+                    .Where(field => !IsBuiltInType(field.Type) && IsComplexFieldAllowedForCreateInput(field, databaseType, definitions))
+                    .Select(field =>
                     {
-                        string typeName = RelationshipDirectiveType.Target(f);
-                        HotChocolate.Language.IHasName? def = definitions.FirstOrDefault(d => d.Name.Value == typeName);
+                        string typeName = RelationshipDirectiveType.Target(field);
+                        HotChocolate.Language.IHasName? def = definitions.FirstOrDefault(d => d.Name.Value.Equals(typeName));
 
                         if (def is null)
                         {
-                            throw new DataApiBuilderException($"The type {typeName} is not a known GraphQL type, and cannot be used in this schema.", HttpStatusCode.InternalServerError, DataApiBuilderException.SubStatusCodes.GraphQLMapping);
+                            throw new DataApiBuilderException(
+                                message: $"The type {typeName} is not a known GraphQL type, and cannot be used in this schema.",
+                                statusCode: HttpStatusCode.ServiceUnavailable,
+                                subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
+                        }
+
+                        if (!entities.TryGetValue(entityName, out Entity? entity) || entity.Relationships is null)
+                        {
+                            throw new DataApiBuilderException(
+                                message: $"Could not find entity metadata for entity: {entityName}.",
+                                statusCode: HttpStatusCode.ServiceUnavailable,
+                                subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
                         }
 
-                        if (def is ObjectTypeDefinitionNode otdn)
+                        string targetEntityName = entity.Relationships[field.Name.Value].TargetEntity;
+                        if (DoesRelationalDBSupportNestedCreate(databaseType) && IsMToNRelationship(entity, field.Name.Value))
```

Comment:
this condition is now redundant: `(DoesRelationalDBSupportNestedCreate(databaseType)` since you have the same on line 86.